lib/types.ts

```diff
@@ -64,21 +75,61 @@ export interface ScannedProjectCustom {
   meta?: any; // not to pollute with actual data; reserved for actual metadata
 }
 
-export enum ScanType {
-  DependencyTree = "DependencyTree",
-  DependencyGraph = "DependencyGraph",
-  ManifestFiles = "ManifestFiles",
+export interface ScanResult {
+  /** TODO: This would be OPTIONAL for ManifestFiles. */
+  artifacts: ScannedArtifact[];
+  /** Not to pollute with actual data; reserved for actual metadata. */
+  meta?: { [key: string]: any };
+}
+
+export interface ScannedArtifact {
+  type: "depTree" | "pkgTree" | "hashes" | "manifestFile";
```

Comment:
`depGraph` should be here too. Perhaps we want to stop using `depTree` or `pkgTree` and only have `depGraph`!